Packet++/src/HttpLayer.cpp

```diff
@@ -448,259 +448,225 @@ void HttpRequestFirstLine::setVersion(HttpVersion newVersion)
-	{"428", HttpResponseLayer::HttpResponseStatusCode::Http428PreconditionRequired },
-	{"429", HttpResponseLayer::HttpResponseStatusCode::Http429TooManyRequests },
-	{"431", HttpResponseLayer::HttpResponseStatusCode::Http431RequestHeaderFieldsTooLarge },
-	{"440", HttpResponseLayer::HttpResponseStatusCode::Http440LoginTimeout },
-	{"444", HttpResponseLayer::HttpResponseStatusCode::Http444NoResponse },
-	{"449", HttpResponseLayer::HttpResponseStatusCode::Http449RetryWith },
-	{"450", HttpResponseLayer::HttpResponseStatusCode::Http450BlockedByWindowsParentalControls },
-	{"451", HttpResponseLayer::HttpResponseStatusCode::Http451 },
-	{"494", HttpResponseLayer::HttpResponseStatusCode::Http494RequestHeaderTooLarge },
-	{"495", HttpResponseLayer::HttpResponseStatusCode::Http495CertError },
-	{"496", HttpResponseLayer::HttpResponseStatusCode::Http496NoCert },
-	{"497", HttpResponseLayer::HttpResponseStatusCode::Http497HTTPtoHTTPS },
-	{"498", HttpResponseLayer::HttpResponseStatusCode::Http498TokenExpiredInvalid },
-	{"499", HttpResponseLayer::HttpResponseStatusCode::Http499 },
-	{"500", HttpResponseLayer::HttpResponseStatusCode::Http500InternalServerError },
-	{"501", HttpResponseLayer::HttpResponseStatusCode::Http501NotImplemented },
-	{"502", HttpResponseLayer::HttpResponseStatusCode::Http502BadGateway },
-	{"503", HttpResponseLayer::HttpResponseStatusCode::Http503ServiceUnavailable },
-	{"504", HttpResponseLayer::HttpResponseStatusCode::Http504GatewayTimeout },
-	{"505", HttpResponseLayer::HttpResponseStatusCode::Http505HTTPVersionNotSupported },
-	{"506", HttpResponseLayer::HttpResponseStatusCode::Http506VariantAlsoNegotiates },
-	{"507", HttpResponseLayer::HttpResponseStatusCode::Http507InsufficientStorage },
-	{"508", HttpResponseLayer::HttpResponseStatusCode::Http508LoopDetected },
-	{"509", HttpResponseLayer::HttpResponseStatusCode::Http509BandwidthLimitExceeded },
-	{"510", HttpResponseLayer::HttpResponseStatusCode::Http510NotExtended },
-	{"511", HttpResponseLayer::HttpResponseStatusCode::Http511NetworkAuthenticationRequired },
-	{"520", HttpResponseLayer::HttpResponseStatusCode::Http520OriginError },
-	{"521", HttpResponseLayer::HttpResponseStatusCode::Http521WebServerIsDown },
-	{"522", HttpResponseLayer::HttpResponseStatusCode::Http522ConnectionTimedOut },
-	{"523", HttpResponseLayer::HttpResponseStatusCode::Http523ProxyDeclinedRequest },
-	{"524", HttpResponseLayer::HttpResponseStatusCode::Http524aTimeoutOccurred },
-	{"598", HttpResponseLayer::HttpResponseStatusCode::Http598NetworkReadTimeoutError },
-	{"599", HttpResponseLayer::HttpResponseStatusCode::Http599NetworkConnectTimeoutError }
+/**
+ * @struct HttpResponseStatusCodeHash
+ * @brief The helper structure for hash HttpResponseStatusCode while using std::unordered_map
+ */
+struct HttpResponseStatusCodeHash
+{
+	size_t operator()(const HttpResponseStatusCode& status) const
+	{
+		return std::hash<int>()(static_cast<int>(status));
```

Comment:
I don't think we need to use `std::hash` here, it's enough to do:

```cpp
size_t operator()(const HttpResponseStatusCode& status) const
{
	return static_cast<int>(status);
}

Reply:
ok